### ISSUE TITLE
Incorrect MRO linearization on valid class

### ISSUE THREAD

**Issue by elprans** [microsoft/pyright#801]
**Describe the bug**
Pyright cannot linearize the MRO of `ObjectType` class, given the following class structure:

```python
class Object:
    pass


class QualifiedObject(Object):
    pass


class DerivableObject(QualifiedObject):
    pass


class SubclassableObject(Object):
    pass


class InheritingObject(DerivableObject, SubclassableObject):
    pass


class Source(QualifiedObject, SubclassableObject):
    pass


class ObjectType(InheritingObject, Source):
    pass
```

**VS Code extension or command-line**

 VS Code, pylance 2020.6.1.

**Comment by erictraut** [comment]
Thanks for the bug report — and your clear and concise repro case.

This will be fixed in the next version of Pyright.

**Comment by erictraut** [comment]
This is addressed in Pyright 1.1.51, which I just published. It will be included in the next release of Pylance.
